scripts/devops_tasks/set_tox_environment.py

```diff
@@ -18,47 +23,77 @@
 PR_BUILD_SET = ["whl", "sdist", "mindependency"]
 
 
-def resolve_devops_variable(var_value):
-    if var_value.startswith("$("):
-        return []
-    else:
-        return [tox_env.strip() for tox_env in var_value.split(",") if tox_env.strip()]
+def resolve_devops_variable(var_value: str) -> List[str]:
+    if var_value:
+        if var_value.startswith("$("):
+            return []
+        else:
+            return [tox_env.strip() for tox_env in var_value.split(",") if tox_env.strip()]
 
 
-def set_devops_value(resolved_set):
+def set_devops_value(resolved_set: List[str]) -> None:
     string_value = ",".join(resolved_set)
-
-    print('Setting environment variable toxenv with value "{}"'.format(string_value))
-    print("##vso[task.setvariable variable=toxenv]{}".format(string_value))
+    set_ci_variable("toxenv", string_value)
 
 
-def remove_unsupported_values(selected_set, unsupported_values):
+def remove_unsupported_values(selected_set: List[str], unsupported_values: List[str]):
     for unsupported_tox_env in unsupported_values:
         if unsupported_tox_env in selected_set:
             selected_set.remove(unsupported_tox_env)
 
 
+def process_ci_skips(service: str) -> None:
+    checks_with_global_skip = ["pylint", "verifywhl", "verifysdist" "bandit", "mypy"]
```

Comment:
Should this list include pyright, verifytypes?

Reply:
The way I was thinking about this originally, I would only apply this logic if the global skip variable is honored. EG we actually have steps that honor `Skip.Pyright`. We _do_ have a step that conditions on `$(Skip.Pylint)`, but we don't on those two you don't mention.

I am going to update this such that it always sets the skip variable, even if it doesn't do anything necessarily.